### PR TITLE
add Country211 prototype dataset

### LINKED ISSUE #5337: Country211
cc @pmeier @bjuncek

### PR DESCRIPTION
**Pull request by Dbhasin1**
Fixes #5337
cc: @pmeier

### CODE REVIEW

File: torchvision/prototype/datasets/_builtin/country211.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+from torchvision.prototype.datasets.utils._internal import hint_sharding, hint_shuffling
+from torchvision.prototype.features import EncodedImage, Label
+
+
+class Country211(Dataset):
+    def _make_info(self) -> DatasetInfo:
+        return DatasetInfo(
+            "country211",
+            homepage="https://github.com/openai/CLIP/blob/main/data/country211.md",
+            valid_options=dict(split=("train", "valid", "test")),
```
Comment (pmeier) [inline torchvision/prototype/datasets/_builtin/country211.py:16]: For consistency with other datasets.

```suggestion
            valid_options=dict(split=("train", "val", "test")),
```

With that we probably need a mapping between the split and the folder name.

File: torchvision/prototype/datasets/_builtin/country211.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        )
+
+    def _filter_split(self, data: Tuple[str, Any], *, split: str) -> bool:
+        return pathlib.Path(data[0]).parent.parent.name == split
+
+    def _make_datapipe(
+        self, resource_dps: List[IterDataPipe], *, config: DatasetConfig
+    ) -> IterDataPipe[Dict[str, Any]]:
+        dp = resource_dps[0]
+        dp = Filter(dp, functools.partial(self._filter_split, split=config.split))
```
Comment (pmeier) [inline torchvision/prototype/datasets/_builtin/country211.py:43]: In this trivial cases, you can avoid an extra method by using `path_comparator` from `torchvision.prototype.datasets.utils._internal`:

```suggestion
        dp = Filter(dp, path_comparator("parent.parent.name", config.split))
```

**Review by pmeier — COMMENTED** [review]
Two minor comments. Otherwise this already looks good @Dbhasin1!

**Review by pmeier — APPROVED** [review]
Thanks @Dbhasin1!

File: torchvision/prototype/datasets/_builtin/country211.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        category = pathlib.Path(path).parent.name
+        return dict(
+            label=Label.from_category(category, categories=self.categories),
+            path=path,
+            image=EncodedImage.from_file(buffer),
+        )
+
+    def _filter_split(self, data: Tuple[str, Any], *, split: str) -> bool:
+        return pathlib.Path(data[0]).parent.parent.name == split
+
```
Comment (pmeier) [inline torchvision/prototype/datasets/_builtin/country211.py:43]: This is no longer needed.

```suggestion
```

**Review by NicolasHug — APPROVED** [review]
Stamping, thanks @Dbhasin1

**Pull request merged**